### ISSUE TITLE
Power BI Action: Publish is returning 'BadRequest' error for Paginated Reports (RDL)

### ISSUE THREAD

**Issue by ldaniel** [maikvandergaag/msft-extensions#328]
Hi everyone, I'm getting a "BadRequest" error when trying to deploy a Paginated Report (RDL) to Power BI Portal, just as shown below. I'm using the task "Upload Power BI report".

Some remarks:
- I don't have the same issue when I deploy a PBIX file, only RDL. 
- I'm also capable of deploying RDL from Power BI Builder with no issues. 
- The App Registration has the proper access in the Workspace.

![Power BI Action publish returning badrequest](https://user-images.githubusercontent.com/205534/140953055-8e468748-e848-4601-a247-d39c46e24abc.png)

**Comment by maikvandergaag** [comment]
@ldaniel: Could you maybe share a test file for me? I will see what I can do.

**Comment by ldaniel** [comment]
hi Maik, sure. I can't share a real report, but the one I'm sharing was built in the Power BI Builder, under the same conditions that the real ones. This report has no dataset or fields, and I getting the same error (BadRequest) as the other ones.

Please, let me know whether this RDL report helps or not. Cheers!

[simple report.zip](https://github.com/maikvandergaag/msft-extensions/files/7507876/simple.report.zip)

**Comment by joaovictormelo42** [comment]
Hi again, @maikvandergaag, i'm sorry to bothering you, but i got the same error:

![image](https://user-images.githubusercontent.com/20947225/141036385-385ca9d5-5c00-4fc2-b4cc-411886296a49.png)

I'm using a service principal to access, and he's got a Admin permission on app.powerbi

If there is something else i can do to better testing this bug.

And if you want, i can close this issue and reopen with this new bug. Again, thanks a lot!

**Comment by maikvandergaag** [comment]
@joaovictormelo42 you have a different error message. You receive a status forbidden back on one of the calls meaning that the spn used does not have the correct permissions.

Would be great if you can open a new issue for this with the complete log. What you could test is running the script locally and see were you lack permissions.

**Comment by ldaniel** [comment]
@maikvandergaag I just found the article below , which I belive can help us on that. After looking to your PowerShell script and see the New-PowerBIReport instruction I ended up finding in the Microsoft Github's powerbi-powershell someone reporting this issue, and a suggested solution was including the extension ".rdl", just like following. Any chances to have this in the plugin? (of course, after you test it)

For RDL
```New-PowerBIReport -Path [FULL PATH TO FILE] -Name [FULL NAME INCLUDING .rdl] -Workspace [WORKSPACE] -ConflictAction Abort```

For PBIX
```New-PowerBIReport -Path [FULL PATH TO FILE] -Name [FRIENDLY NAME] -Workspace [WORKSPACE] -ConflictAction CreateOrOverwrite```

Source: https://github.com/microsoft/powerbi-powershell/issues/319

**Comment by maikvandergaag** [comment]
Yes that is what 8 thought as well. There are two option the cmd commands from Microsoft or the API. Will first do some testing with the cmd and if that does not work I will use the API.

Hope to have something in beta at the end of the week

**Comment by mattias-de-coninck** [comment]
@maikvandergaag from personal experience I know that uploading RDLs can cause some issues with the default CmdLets.
However the import APIs https://docs.microsoft.com/en-us/rest/api/power-bi/imports/post-import-in-group should be able to handle this.

**Comment by maikvandergaag** [comment]
@ldaniel I have a version that is now beign deployed to beta that contains the functionality. I only do not have premium workspace to test. To you have an option to test the beta?

If yes could you please share your Azure DevOps organization name so I can give you access to the beta

**Comment by ldaniel** [comment]
@maikvandergaag great news! Yes, I'll send you, in private.

**Comment by ldaniel** [comment]
@maikvandergaag many thanks, the Power BI Paginated Reports (RDL) are now supported in this plugin! Great achievement!👍🏼

**Comment by tpaquin9** [comment]
@maikvandergaag I'm also interested in leveraging this extension to deploy RDL files. Is this change still in beta at this time?

**Comment by maikvandergaag** [comment]
NO it should work with the current production version

**Comment by thomaspaquin** [comment]
@maikvandergaag Thanks! Looks like the reason it wasn't working for me is that I'm incorporating a hierarchical naming structure. I.e. All reports are named "[Parent Group] Report Name" Which displays each report as a child element of the parent. At a glance, it seems like the api doesn't like the presence of brackets in the file path, but I'll see if I can find a workaround.

Thanks for your help!

**Comment by llopezalonso** [comment]
Hi @maikvandergaag 
We are experiencing the 400-BadRequest issue with the version 5.4.31 of the extension:

![image](https://user-images.githubusercontent.com/40022962/161024797-8ef7518d-ecf8-4252-8adb-a504952dd518.png)

We have change our rdl name with POC.rdl but it does not work for us.

Could you help us?

Thanks a ton for your support!

**Comment by maikvandergaag** [comment]
Please share more info about the issue

**Comment by llopezalonso** [comment]
We are connecting to Power BI through a SPN who was added through Power BI Admin Portal. Our SPN is added to the workspace as an Admin, we have no problems uploading PBI files, but we are experiencing the 400-BadRequest with RDL's.

**Comment by maikvandergaag** [comment]
could you share the log with the variable system.debug set to true

**Comment by llopezalonso** [comment]
@maikvandergaag sent in private

**Comment by llopezalonso** [comment]
Hi @maikvandergaag did you receive the requested log?

Thanks a ton for your support :)

Regards,
Luis López Alonso

**Comment by maikvandergaag** [comment]
Sorry for the late response. Log is mentioning that it will abort on already existing content do you already have content with the same name in the workspace?

**Comment by llopezalonso** [comment]
Hi @maikvandergaag ,

Apologizes for the delayed response.

This RDL was already uploaded and we experienced this error. Our next step was to update the RDL name to a different one to see if it was the problem, but we experienced the same issue.

**Comment by maikvandergaag** [comment]
Could you share a updated log file?
